chrome/test/chrome-api-shim.js

```diff
@@ -12,15 +10,17 @@ var chrome = {};
   }
 
   chrome.storage = {};
-  chrome.storage.sync = {
-    "set": function(dict, callback) {
+  function Storage() {
+    var storage_ = {};
+
+    this.set = function(dict, callback) {
```

Comment:
Ok. The difference is that in my approach storage_ was be private (hidden in a closure), although that doesn't matter much for our purposes.